lib/prometheus/client/data_stores/direct_file_store.rb

```diff
@@ -182,13 +185,15 @@ def process_id
 
           def aggregate_values(values)
             if @values_aggregation_mode == SUM
-              values.inject { |sum, element| sum + element }
+              values.map { |element| element[0] }.inject { |sum, value| sum + value }
```

Comment:
For all 3 of these, could we `map(&:first)` and then do the same thing we were doing before?
That feels a lot more readable to me.

(so: `values.map(&:first).inject { |sum, element| sum + element }`, `values.map(&:first).max`, etc)
I'm half wondering why we did this `inject` before, instead of just calling `.sum`... Is there a large performance penalty to `map`? Could we measure that?
The reason I mention that is that we already have some users with performance problems on scrape, and i'd thus like to make this as fast as possible.

If `map` is much slower, we should probably change this first one to the same inject as before, but the body of the block to `sum + element[0]`...
And i'm not sure about the ones below, whether that's the fastest way?

Reply:
I am very open to changing this code. I mostly just wanted to get it working without considering performance that much.